### PR TITLE
Update cron tests

### PR DESCRIPTION
**Pull request by namurphy**
- I changed weekly tests to approximately fortnightly since that seemed a more appropriate cadence, and because it's an excuse to use the word "fortnightly".  
 - I added tests of the development versions of matplotlib (inspired by #1330) and SciPy so that we can catch any API changes before official releases.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -12,10 +12,12 @@ setenv =
     PYTEST_COMMAND = pytest --pyargs plasmapy --durations=25 -n=auto --dist=loadfile
 extras = tests
 deps =
-    numpydev: git+https://github.com/numpy/numpy
     astropydev: git+https://github.com/astropy/astropy
-    xarraydev: git+https://github.com/pydata/xarray
+    matplotlibdev: git+https://github.com/matplotlib/matplotlib
+    numpydev: git+https://github.com/numpy/numpy
+    scipydev: git+https://github.com/matplotlib/matplotlib
```
Comment (StanczakDominik) [inline tox.ini:18]: Okay, first up that last one points to matplotlib rather than scipy :) Secondly, and this is something I've been meaning to improve for a while - this way of installing these three packages means we're building huge binary blobs from scratch, wasting time and energy.

Instead, we can leverage prebuilt wheels as per [sklearn docs](https://scikit-learn.org/stable/developers/advanced_installation.html#installing-nightly-builds): something like
```bash
pip install --pre --extra-index https://pypi.anaconda.org/scipy-wheels-nightly/simple scikit-learn
````
should do the trick.

In fact, we can replace all three of these, since they're listed under https://anaconda.org/scipy-wheels-nightly/repo?type=pypi&label=main.

We just need to figure out whether `tox deps` will take this kind of argument. Checking...

**Review by StanczakDominik — CHANGES_REQUESTED** [review]
Looks great but for a possible optimization!

File: tox.ini (hunk 1 above)
Comment (StanczakDominik) [reply tox.ini]: ... for some reason I saw matplotlib where it wasn't there. I've pushed a commit to try this out.

**Review by StanczakDominik — APPROVED** [review]

**Comment by StanczakDominik** [comment]
Well, of course the pushed commit won't run these tests on the PR; but I'm happy to merge this and try it out on the field of battle.

File: tox.ini (hunk 1 above)
Comment (namurphy) [reply tox.ini]: That's a really good idea!  Being able to run these tests quickly if definitely helpful, especially since we'd need to trigger runs occasionally to see if things get fixed.

**Review by StanczakDominik — CHANGES_REQUESTED** [review]
Stop the presses, actually, I'm having trouble with scipy due to a version conflict:

<details>
<summary>Details</summary>

```
🕙[ 19:20:44 ] ➜ tox -e py39-scipydev --notest
py39-scipydev create: /home/dominik/Code/github/PlasmaPy/PlasmaPy/.tox/py39-scipydev
py39-scipydev installdeps: :NIGHTLY:scipy, pytest-xdist, pytest-github-actions-annotate-failures
ERROR: invocation failed (exit code 1), logfile: /home/dominik/Code/github/PlasmaPy/PlasmaPy/.tox/py39-scipydev/log/py39-scipydev-1.log
===================================== log start ======================================
Looking in indexes: https://pypi.anaconda.org/scipy-wheels-nightly/simple
Collecting scipy
  Downloading https://pypi.anaconda.org/scipy-wheels-nightly/simple/scipy/1.8.0.dev0%2B2083.8dd161d/scipy-1.8.0.dev0%2B2083.8dd161d-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (102.6 MB)
  Downloading https://pypi.anaconda.org/scipy-wheels-nightly/simple/scipy/1.8.0.dev0%2B2082.296b550/scipy-1.8.0.dev0%2B2082.296b550-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (102.6 MB)
  Downloading https://pypi.anaconda.org/scipy-wheels-nightly/simple/scipy/1.8.0.dev0%2B2060.9484b9d/scipy-1.8.0.dev0%2B2060.9484b9d-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (104.8 MB)
  Downloading https://pypi.anaconda.org/scipy-wheels-nightly/simple/scipy/1.8.0.dev0%2B2047.048bac7/scipy-1.8.0.dev0%2B2047.048bac7-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (104.8 MB)
  Downloading https://pypi.anaconda.org/scipy-wheels-nightly/simple/scipy/1.8.0.dev0%2B1984.f1c6b54/scipy-1.8.0.dev0%2B1984.f1c6b54-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (105.2 MB)
  Downloading https://pypi.anaconda.org/scipy-wheels-nightly/simple/scipy/1.8.0.dev0%2B1981.16d9e82/scipy-1.8.0.dev0%2B1981.16d9e82-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (104.7 MB)
  Downloading https://pypi.anaconda.org/scipy-wheels-nightly/simple/scipy/1.8.0.dev0%2B1869.838cfbe/scipy-1.8.0.dev0%2B1869.838cfbe-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (102.4 MB)
  Downloading https://pypi.anaconda.org/scipy-wheels-nightly/simple/scipy/1.8.0.dev0%2B1832.bdf109c/scipy-1.8.0.dev0%2B1832.bdf109c-cp39-cp39-manylinux_2_5_x86_64.manylinux1_x86_64.whl (98.3 MB)
  Downloading https://pypi.anaconda.org/scipy-wheels-nightly/simple/scipy/1.8.0.dev0%2B1776.48e1dd7/scipy-1.8.0.dev0%2B1776.48e1dd7-cp39-cp39-manylinux_2_5_x86_64.manylinux1_x86_64.whl (98.3 MB)
  Downloading https://pypi.anaconda.org/scipy-wheels-nightly/simple/scipy/1.8.0.dev0%2B1761.be2027d/scipy-1.8.0.dev0%2B1761.be2027d-cp39-cp39-manylinux_2_5_x86_64.manylinux1_x86_64.whl (98.3 MB)
  Downloading https://pypi.anaconda.org/scipy-wheels-nightly/simple/scipy/1.8.0.dev0%2B1753.a063cf3/scipy-1.8.0.dev0%2B1753.a063cf3-cp39-cp39-manylinux_2_5_x86_64.manylinux1_x86_64.whl (107.5 MB)
ERROR: Cannot install scipy==1.8.0.dev0+1753.a063cf3, scipy==1.8.0.dev0+1761.be2027d, scipy==1.8.0.dev0+1776.48e1dd7, scipy==1.8.0.dev0+1832.bdf109c, scipy==1.8.0.dev0+1869.838cfbe, scipy==1.8.0.dev0+1981.16d9e82, scipy==1.8.0.dev0+1984.f1c6b54, scipy==1.8.0.dev0+2047.048bac7, scipy==1.8.0.dev0+2060.9484b9d, scipy==1.8.0.dev0+2082.296b550 and scipy==1.8.0.dev0+2083.8dd161d because these package versions have conflicting dependencies.

The conflict is caused by:
    scipy 1.8.0.dev0+2083.8dd161d depends on numpy>=1.17.3
    scipy 1.8.0.dev0+2082.296b550 depends on numpy>=1.17.3
    scipy 1.8.0.dev0+2060.9484b9d depends on numpy>=1.17.3
    scipy 1.8.0.dev0+2047.048bac7 depends on numpy>=1.17.3
    scipy 1.8.0.dev0+1984.f1c6b54 depends on numpy>=1.17.3
    scipy 1.8.0.dev0+1981.16d9e82 depends on numpy>=1.17.3
    scipy 1.8.0.dev0+1869.838cfbe depends on numpy>=1.17.3
    scipy 1.8.0.dev0+1832.bdf109c depends on numpy>=1.17.3
    scipy 1.8.0.dev0+1776.48e1dd7 depends on numpy>=1.17.3
    scipy 1.8.0.dev0+1761.be2027d depends on numpy>=1.17.3
    scipy 1.8.0.dev0+1753.a063cf3 depends on numpy>=1.17.3

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/user_guide/#fixing-conflicting-dependencies

====================================== log end =======================================
ERROR: could not install deps [:NIGHTLY:scipy, pytest-xdist, pytest-github-actions-annotate-failures]; v = InvocationError('/home/dominik/Code/github/PlasmaPy/PlasmaPy/.tox/py39-scipydev/bin/python -m pip install -i https://pypi.anaconda.org/scipy-wheels-nightly/simple scipy', 1)
______________________________________ summary _______________________________________
ERROR:   py39-scipydev: could not install deps [:NIGHTLY:scipy, pytest-xdist, pytest-github-actions-annotate-failures]; v = InvocationError('/home/dominik/Code/github/PlasmaPy/PlasmaPy/.tox/py39-scipydev/bin/python -m pip install -i https://pypi.anaconda.org/scipy-wheels-nightly/simple scipy', 1)
```

</details>

**Comment by StanczakDominik** [comment]
Hahahah, good timing! 🤣 

I'll fix this up in a follow up PR.

**Comment by namurphy** [comment]
> Hahahah, good timing! rofl
> 
> I'll fix this up in a follow up PR.

Ah, oops!  Thanks for the follow-up!
